recipes/filepattern/meta.yaml

```
@@ -0,0 +1,97 @@
+{% set name = "filepattern" %}
+{% set version = "2.0.1" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }} 
+
+source:
+  url: https://github.com/PolusAI/filepattern/archive/v{{ version }}.tar.gz
+  sha256: d5129345fc46bba18716757ea64b3783ce1a033aa81a437f5030a30bcdc670a7
+
+build:
+  skip: true  # [py2k]
+  number: 0
+
+requirements:
+  build:
+    - cmake
+    - make  # [not win]
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - git
+
+outputs:
+  - name: filepattern
+    script: build.sh  # [unix]
+    script: bld.bat  # [win]
+    version: {{ version }}
+    requirements:
+      build:
+        - cmake
+        - make  # [not win]
+        - {{ compiler('c') }}
+        - {{ compiler('cxx') }}
+        - git
+      host:
+        - python
+        - setuptools
+        - pip
+        - pybind11
```

```suggestion
      host:
        - python
        - setuptools
        - pip
        - pybind11
        - libfilepattern {{ version }}
```
Shouldn't this package depend on libfilepattern?

---

we are actually creating two separate packages. One python package (`filepattern`) and another C++ package `libfilepattern`. Should we create two separate recipes or use one to create both packages?

---

@carterbox : Trying to get some clarification here. Our intention is to create two separate things. One `filepattern` Python package and a `libfilepattern` C++ library. Should we split this recipe into two to do that? Both the Python package and C++ library uses the same source code but different CMake options to produce them. There is no dependency between them.

---

The python package should link dynamically to the c++ shared library and the two packages should live in separate feedstocks/recipes in order to reduce the number of redundant builds.

---

We have created a new recipe (https://github.com/conda-forge/staged-recipes/pull/24433) for libfilepattern. We also updated the build process for the Python package so that if `libfilepattern` is present, it will use that instead of compiling the core package files. Once, the `libfilepattern` recipe is merged, we will rework this recipe.

---

@carterbox : we updated the recipe to depend on the C++ shared library (as suggested). Can you please take a look at see if it is good to be merged? Thanks :-)